### PR TITLE
chore: update controller-gen to v0.9.2

### PR DESCRIPTION
**Pull request by jan--f**
This mostly pulls in many fixes from controller-tools. Mainly
controller-gen now respects type marker in types and generated the CRD
manifests accordingly. This means that SSA now functions correctly with
the generated manifests. Accordingly this also reenables the
`revert-test`.
The other changes added here are mostly whitespace changes in the
manifests, the status being no longer included in the CRD manifests (see
https://github.com/kubernetes-sigs/controller-tools/issues/456 for
details) and we seem to have forgotten to regenerate some manifests
after a version bump.

Fixes: https://github.com/rhobs/observability-operator/issues/142

Signed-off-by: Jan Fajerski <jfajersk@redhat.com>

### CODE REVIEW

**Review by simonpasquier — APPROVED** [review]

**Comment by sthaha** [comment]
lgtm except for the broken CI 🤔

**Comment by jan--f** [comment]
Ok the CRD manifest errors where due to a bug in controller-gen. v0.9.2 ships a fix for this.

**Review by simonpasquier — DISMISSED** [review]

File: hack/kind/setup.sh [hunk 1]
```diff
@@ -77,11 +77,23 @@ setup_cluster() {
...
 }
 
+install_kubectl() {
+    if ! command -v kubectl &> /dev/null
+then
+    info "kubectl not found, attempting to install"
+    curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+    chmod +x kubectl
+    export PATH="$(pwd):$PATH"
+fi
```
Comment (sthaha) [inline hack/kind/setup.sh:87]: Unrelated change :( 

Also ... IMHO, we shouldn't install tools / make changes to users system other than in locations known/announced to them. E.g. `make tools` is explicit and installs to `<prj>/tmp/bin`

**Review by sthaha — CHANGES_REQUESTED** [review]
Requesting changes only due to an unrelated change (`hack/setup.sh`). 
Merging this would squash the commits (only policy enabled for this repo)

**Comment by jan--f** [comment]
> Requesting changes only due to an unrelated change (`hack/setup.sh`). Merging this would squash the commits (only policy enabled for this repo)

Thanks good catch. I had that change in for testing. I'm actually proposing it in #175. Dropped it now.

**Review by sthaha — APPROVED** [review]

**Pull request merged**